[10.0][FIX] website_sale_charge_payment_fee: Use a browse record if there a…

SimoRubi opened the pull request:
…re no acquirers

SimoRubi committed:
[FIX] website_sale_charge_payment_fee: Use a browse record if there are no acquirers

SimoRubi commented:
@OCA/e-commerce-maintainers can someone please take a look? Thanks

SimoRubi commented:
@OCA/e-commerce-maintainers can someone please take a look?
How can I help this PR to be merged?
Thanks

simahawk commented on website_sale_charge_payment_fee/controllers/website_sale.py:
```diff
@@ -25,7 +25,9 @@ def payment(self, **post):
-                selected_acquirer = values['acquirers'][0]
+                selected_acquirer = request.env['payment.acquirer'].browse()
+                if values['acquirers']:
+                    selected_acquirer = values['acquirers'][0]
```
I don't know this module but I'm a bit sceptic about this change. The original module provides a list of rendered HTML, hence what you would have here is not an ID or a recordset and the template does seem to take this into account.
Am I wrong?

See https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L747

SimoRubi replied:
Hi, thanks for checking!
Actually the original module provides a list of records found with `search` https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L744-L746
and then appended one by one to `value['acquirers]` in https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L760; the rendered HTML is just assigned to the `button` field of every acquirer in the list.

simahawk replied:
hmm seems you are right. Sorry, I probably checked too quickly :sweat: . Then I don't see any blockers...

simahawk reviewed (approved):
reminder: version update is not needed any more ;)

SimoRubi commented:
@OCA/e-commerce-maintainers how can I help this PR to be merged? Thanks

simahawk commented:
/ocabot merge

simahawk commented:
@SimoRubi tests are failing there, can you check?

SimoRubi commented:
@simahawk rebased on `10.0`.
Tests are failing because they are already failing in `10.0`:
![image](https://user-images.githubusercontent.com/32064796/74028663-31164600-49ab-11ea-9be1-dff7c1618628.png)
(from https://travis-ci.org/OCA/e-commerce/branches)

SimoRubi commented:
@simahawk does this PR have to wait for the fix of `10.0`?
If no-one is tackling it, I can at least try.

simahawk commented:
@SimoRubi 10.0 is green https://github.com/OCA/e-commerce/commits/10.0

![image](https://user-images.githubusercontent.com/347149/74220383-bcdfe900-4caf-11ea-8dfd-3a75b4347164.png)

What you see there is an old build. The green one is pretty old and might fail today.

In your build I see `raise UserError(_("Unable to find Wkhtmltopdf on this system. The PDF can not be created."))`.

You can rebase on 10.0 and see if it happens again.

SimoRubi commented:
@simahawk the last commit is _Unverified_ because it has [ci skip]:
![image](https://user-images.githubusercontent.com/32064796/74220637-61622b00-4cb0-11ea-9405-4d71bc545ce4.png)
That means that Travis did not run on that commit, you can verify that by clicking on the :heavy_check_mark: :
![image](https://user-images.githubusercontent.com/32064796/74220702-8ce51580-4cb0-11ea-9efa-2aa6f6111c25.png)

The last _verified_ commit is the previous one, that is red:
![image](https://user-images.githubusercontent.com/32064796/74220780-c453c200-4cb0-11ea-9238-ccba07a8ecf2.png)
and its travis build corresponds to the latest build of branch `10.0` that is https://travis-ci.org/OCA/e-commerce/builds/564785979 and is failed.

Anyway I already rebased like I said in https://github.com/OCA/e-commerce/pull/264#issuecomment-583364617 and from that moment on, tests began failing in this PR too (they were green before, when you first tried to merge).

simahawk commented:
@SimoRubi oh yeah, sorry (repeat with me: don't do 3 things at the same time when answering to ppl :sweat_smile: )
Then... you know what to do: try to fix 10.0 :)

SimoRubi commented:
> @SimoRubi oh yeah, sorry (repeat with me: don't do 3 things at the same time when answering to ppl  )
> Then... you know what to do: try to fix 10.0 :)

@simahawk I've fixed `10.0` in https://github.com/OCA/e-commerce/pull/366 and rebased, can you try again to merge?
Thanks!

simahawk commented:
/ocabot merge nobump
